### ISSUE TITLE
Remove spec for in-memory exporters

### ISSUE THREAD

**Issue by alanwest** [open-telemetry/opentelemetry-specification#2425]
Currently, the spec considers the in-memory exporter a push metric exporter:

> In-memory Metrics Exporter is a [Push Metric Exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#push-metric-exporter) which accumulates metrics data in the local memory and allows to inspect it (useful for e.g. unit tests).

The original intent of #2415 was to define a default interval to push metrics from a periodic metric reader paired with the in-memory exporter. The interval I proposed was infinite - that is, by default, an explicit force flush would be required to collect metrics. This raised the question of whether the in-memory exporter is actually a pull metric exporter.

An argument was offered in favor of leaving it as a push exporter https://github.com/open-telemetry/opentelemetry-specification/pull/2415#issuecomment-1068327497.

A number of folks felt that providing it as pull metric exporter offers better ergonomics for the primary use case which is unit tests. One suggestion (https://github.com/open-telemetry/opentelemetry-specification/pull/2415#issuecomment-1069335024) is that both push and pull may be valuable.

**Comment by aabmass** [comment]
I like the suggestion that we don't specify if its push or pull which @alanwest mentioned.

**Comment by alanwest** [comment]
> I like the suggestion that we don't specify if its push or pull which @alanwest mentioned.

I was taking this a step further and questioning whether we can just remove the entire in-memory exporter specification.

In contrast, I see value in the stdout exporter specification since it's pretty common for end users to pick it up when experimenting with the OTel SDKs. It makes sense to me that it has some semblance of uniform usage/behavior from language to language.

I'm uncertain if striving for uniformity should also be a requirement for the in-memory exporter as I'm unfamiliar with any scenarios where end users are using it.

**Comment by ocelotl** [comment]
> I was taking this a step further and questioning whether we can just remove the entire in-memory exporter specification.

Yes, I agree. The spec should not be concerned with debugging or testing tools. Both of them can be very language specific:
- In the case of debugging, what this exporter can do, a debugger can do as well in many cases.
- In the case of testing, what this exporter can do, a mock can do as well in many cases.

If this is still something that others consider should be part of the spec, at least make this optional.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> The premise no longer exists: the current specification/metrics/sdk.md (fetched today) has no 'In-memory Metrics Exporter' section and no push/pull classification of it (only generic 'in-memory state' diagram labels remain). In-thread, maintainers aabmass, alanwest, and ocelotl all favored removing/not specifying the in-memory exporter, which is what happened, so the push-vs-pull question is moot.

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> Verified obsolete: current specification/metrics/sdk.md has no "In-memory Metrics Exporter" section and no push/pull classification of it (only generic "In-memory state" diagram labels at lines ~1426-1782); the quoted "In-memory Metrics Exporter is a Push Metric Exporter" text is gone. Maintainers aabmass, alanwest, ocelotl all favored not-specifying/removing it (#issuecomment-1072847519, -1072894587, -1074759744), which happened, mooting the push-vs-pull ask. Issue still OPEN but timeline shows no closing PR/duplicate/cross-ref (only labeled/commented/mentioned/assigned events).

Claude evaluated this issue with high confidence that it can be closed.

Claude has classified this issue as obsolete.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.

**Comment by alanwest** [comment]
Hey @ocelotl, you pinged me on Slack about this issue.

I do not have strong feelings whether this issue remain open or not, but I think Claude overlooked the presence of this spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/in-memory.md.

It's been a long time, but I think what I was proposing here was that the in-memory spec be deleted.

**Comment by ocelotl** [comment]
Right, @alanwest thanks for taking a look here. I think this issue should not be closed, the intention here is to remove the in memory exporter from the spec. I suggest the issue be renamed to reflect this.

**Comment by svrnm** [comment]
keeping this open for now, maybe worth to bring it back to a spec meeting?
